quicklogic/common/toolchain_wrappers/symbiflow_synth

```diff
@@ -5,114 +5,102 @@ MYPATH=`realpath $0`
+    echo "                       [+define+<macro name>[=<macro value>]]"
+    echo "                       [-f <additional compile command file>]"
+    echo ""
+    echo "note: device and part parameters are required if xdc is passed"
+    exit 1
+}
 
 VERILOG_FILES=()
 XDC_FILES=()
-TOP=top
+TOP="top"
 DEVICE=""
 FAMILY=""
 PART=""
 PCF=""
+EXTRA_ARGS=()
 
-VERILOGLIST=0
-XDCLIST=0
-TOPNAME=0
-DEVICENAME=0
-FAMILYNAME=0
-PARTNAME=0
-PCFNAME=0
-
+OPT=""
 for arg in $@; do
-	echo $arg
-	case "$arg" in
-		-t|--top)
-			echo "adding top"
-			VERILOGLIST=0
-			XDCLIST=0
-			TOPNAME=1
-			DEVICENAME=0
-			FAMILYNAME=0
-			PARTNAME=0
-			PCFNAME=0
-			;;
-		-x|--xdc)
-			VERILOGLIST=0
-			XDCLIST=1
-			TOPNAME=0
-			DEVICENAME=0
-			FAMILYNAME=0
-			PARTNAME=0
-			PCFNAME=0
-			;;
-		-v|--verilog)
-			VERILOGLIST=1
-			XDCLIST=0
-			TOPNAME=0
-			DEVICENAME=0
-			FAMILYNAME=0
-			PARTNAME=0
-			PCFNAME=0
-			;;
-		-d|--device)
-			VERILOGLIST=0
-			XDCLIST=0
-			TOPNAME=0
-			DEVICENAME=1
-			FAMILYNAME=0
-			PARTNAME=0
-			PCFNAME=0
-			;;
-		-f|--family)
-			VERILOGLIST=0
-			XDCLIST=0
-			TOPNAME=0
-			DEVICENAME=0
-			FAMILYNAME=1
-			PARTNAME=0
-			PCFNAME=0
-			;;
-		-P|--part)
-			VERILOGLIST=0
-			XDCLIST=0
-			TOPNAME=0
-			DEVICENAME=0
-			FAMILYNAME=0
-			PARTNAME=1
-			PCFNAME=0
-			;;
-		-p|--pcf)
-			VERILOGLIST=0
-			XDCLIST=0
-			TOPNAME=0
-			DEVICENAME=0
-			FAMILYNAME=0
-			PARTNAME=0
-			PCFNAME=1
-			;;
-		*)
-			if [ $VERILOGLIST -eq 1 ]; then
-				VERILOG_FILES+=($arg)
-			elif [ $XDCLIST -eq 1 ]; then
-				XDC_FILES+=($arg)
-			elif [ $TOPNAME -eq 1 ]; then
-				TOP=$arg
-			elif [ $DEVICENAME -eq 1 ]; then
-				DEVICE=$arg
-			elif [ $FAMILYNAME -eq 1 ]; then
-				FAMILY=$arg
-			elif [ $PARTNAME -eq 1 ]; then
-				PART=$arg
-			elif [ $PCFNAME -eq 1 ]; then
-				PCF=$arg
-			else
-				echo "Usage: synth [-t|--top <top module name> -v|--verilog <Verilog files list> [-x|--xdc <XDC files list>]"
-				echo "             [-d|--device <device type (e.g. artix7)>] [-P|--part <part name>]"
-				echo "             [-f|--family <device family (pp3,ap3)>] [-P|--part <part name>]"
-				echo "note: device and part parameters are required if xdc is passed"
-				exit 1
-			fi
-			;;
-	esac
+    case $arg in
+        -t|--top)
+            OPT="top"
+            ;;
+#        -x|--xdc)
+#            OPT="xdc"
+#            ;;
+        -v|--verilog)
+            OPT="vlog"
+            ;;
+        -d|--device)
+            OPT="dev"
+            ;;
+        -F|--family)
+            OPT="family"
+            ;;
+        -P|--part)
+            OPT="part"
+            ;;
+        -p|--pcf)
+            OPT="pcf"
+            ;;
+        -y|-f|+incdir+*|+libext+*|+define+*)
+            OPT="xtra"
+            ;;
+        *)
+            case $OPT in
+                "top")
+                    TOP=$arg
+                    OPT=""
+                    ;;
+#                "xdc")
+#                    XDC_FILES+=($arg)
+#                    ;;
```

Comment:
As before, I guess this can just be removed

Reply:
Fixed